### PR TITLE
Implement configurable storage backend

### PR DESCRIPTION
**Pull request by viralpraxis**
Resolves https://github.com/huacnlee/rails-settings-cached/issues/234

Configuration example:
```ruby
# config/initializers/rails_settings_cached.rb

RailsSettings.configure do
  self.storage = ActiveSupport::Cache::RedisCacheStore.new(url: "redis://localhost:6379")
end

```

### CODE REVIEW

**Comment by huacnlee** [comment]
I'd like to accept these changes.

1. Please write a test.
2. This `storage` must have a default storage.

File: lib/rails-settings/base.rb [hunk 1]
```diff
@@ -127,7 +127,7 @@ def rails_initialized?
       end
 
       def _all_settings
-        RequestCache.all_settings ||= Rails.cache.fetch(cache_key, expires_in: 1.week) do
+        RequestCache.all_settings ||= RailsSettings.configuration.storage.fetch(cache_key, expires_in: 1.week) do
```
Comment (huacnlee) [inline lib/rails-settings/base.rb:130]: Create a method for use.

```rb
def storage
  RailsSettings.configuration.storage.
end
```

File: lib/rails-settings/configuration.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          Option `storage` must be an instance of `ActiveSupport::Cache::Store` class.
+        TXT
+      end
+
+      @storage = value
+    end
+  end
+
+  class << self
+    def configuration
```
Comment (huacnlee) [inline lib/rails-settings/configuration.rb:25]: Rename to `config`, like Rails or other RubyGems.

Comment (huacnlee) [reply lib/rails-settings/configuration.rb:25]: https://github.com/huacnlee/rucaptcha/blob/main/lib/rucaptcha/configuration.rb
https://github.com/huacnlee/rucaptcha/blob/main/lib/rucaptcha.rb#L23

Please follow this code to define a configuration.

File: lib/rails-settings/base.rb (hunk 1 above)
Comment (viralpraxis) [reply lib/rails-settings/base.rb]: Done

File: lib/rails-settings/configuration.rb (hunk 2 above)
Comment (viralpraxis) [reply lib/rails-settings/configuration.rb]: Done

**Comment by viralpraxis** [comment]
Tests are fixed (hopefully)

**Comment by viralpraxis** [comment]
@huacnlee seems like you used outdated configuration API version here: https://github.com/huacnlee/rails-settings-cached/commit/ed2ab02b3611ae36c5a509742b6547d883df40b1

Current implementation assumes this kind of usage:

```ruby
RailsSettings.configure do
  self.cache_storage = ActiveSupport::Cache::RedisCacheStore.new(url: "redis://localhost:6379")
end
```

**Pull request merged**